bpf/tests/tc_egressgw_redirect_from_host.c

```diff
@@ -213,6 +216,214 @@ int egressgw_drop_no_egress_ip_check(const struct __ctx_buff *ctx)
 
 	del_egressgw_policy_entry(CLIENT_IP, EXTERNAL_SVC_IP, 32);
 	endpoint_v4_del_entry(GATEWAY_NODE_IP);
+	map_delete_elem(&METRICS_MAP, &key);
+
+	test_finish();
+}
+
+/* Test that a packet matching an egress gateway policy on the to-netdev
+ * program gets redirected to the gateway node (IPv6).
+ */
+PKTGEN("tc", "tc_egressgw_redirect_v6")
+int egressgw_redirect_pktgen_v6(struct __ctx_buff *ctx)
+{
+	return egressgw_pktgen_v6(ctx, (struct egressgw_test_ctx) {
+			.test = TEST_REDIRECT,
+	});
+}
+
+SETUP("tc", "tc_egressgw_redirect_v6")
+int egressgw_redirect_setup_v6(struct __ctx_buff *ctx)
+{
+	union v6addr ext_svc_ip = EXTERNAL_SVC_IP_V6;
+	union v6addr client_ip = CLIENT_IP_V6;
+	union v6addr egress_ip = EGRESS_IP_V6;
+
+	ipcache_v6_add_entry(&ext_svc_ip, 0, WORLD_IPV6_ID, 0, 0);
+	create_ct_entry_v6(ctx, client_port(TEST_REDIRECT));
+	add_egressgw_policy_entry_v6(&client_ip, &ext_svc_ip, 48, GATEWAY_NODE_IP, &egress_ip);
```

Comment:
Besides me being inconsistent using /48 and /64 prefixes in the different egw BPF testsuites :grimacing:, I think I should change all of them to /64. IPv4 testsuites mostly use /24, not sure what the best equivalent is.

Reply:
I wouldn't sweat it. If you add a macro & use it in all the entries, it's easy enough to change later if needed.